File: tests/parsing/rust/numeric_underscores.rs
// Underscores as visual separators in numeric literals (Rust RFC 3101).
// Exercises Parsed_int for radix-prefix underscores (e.g. 0x_dead_beef) and
// between-digit separators.

fn decimal() {
    let _ = 1_000_000;
    let _ = 98_222;
}

fn hex_oct_bin() {
    let _ = 0x_dead_beef;
    let _ = 0xDEAD_BEEF;
    let _ = 0x_1;
    let _ = 0o_755;
    let _ = 0b_1010_1010;
}

// Rust allows multiple underscores after the radix prefix (unlike Python).
fn multiple_underscores_after_prefix() {
    let _ = 0b________1;
}

File: tests/parsing/python/numeric_underscores.py
# PEP 515 underscores as visual separators in numeric literals (Python 3.6).

# Decimal int.
a = 1_000_000
b = 10_000_000_000

# Hexadecimal, octal, binary.
c = 0x_dead_beef
d = 0xDEAD_BEEF
e = 0o_755
f = 0b_1010_1010

# Float with fractional, exponent, and underscores in both parts.
g = 1_000.5
h = 1_000.000_5
i = 1_000.5e1_0
j = 1.5e-1_0

# Complex / imaginary literal.
k = 1_000j
m = 1_000.5j
